src/diffusers/models/controlnet_sd3.py

```diff
@@ -241,9 +247,10 @@ def _set_gradient_checkpointing(self, module, value=False):
         if hasattr(module, "gradient_checkpointing"):
             module.gradient_checkpointing = value
 
+    # If controlnet is used for inpainting, the num_extra_conditioning_channels should be 1.
     @classmethod
     def from_transformer(
-        cls, transformer, num_layers=12, num_extra_conditioning_channels=1, load_weights_from_transformer=True
+        cls, transformer, num_layers=12, num_extra_conditioning_channels=0, load_weights_from_transformer=True
```

Comment:
do we have to change the default? you can manually pass `num_extra_conditioning_channels=0` when you need it to be `0`, no?

otherwise I will tag the original author to see if it is ok to change here (to make sure it won't break anything)

Reply:
Because the method in training [script ](https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet_sd3.py#L986) use the default parameters and the script is for add control with normal image instead of inpainting.  As the result, with num_extra_conditioning_channels=0 in this method, [script ](https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet_sd3.py#L986) will occur errors.
In addtion, the default value of num_extra_conditioning_channels in [__ini__](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/controlnets/controlnet_sd3.py#L58) method is also 0.